editor/editor_node.cpp

```diff
@@ -2741,47 +2741,42 @@ Dictionary EditorNode::_get_main_scene_state() {
 	state["property_edit_offset"] = get_property_editor()->get_scene_tree()->get_vscroll_bar()->get_value();
 	state["saved_version"] = saved_version;
 	state["node_filter"] = scene_tree_dock->get_filter();
+	return state;
+}
+
+void EditorNode::_set_main_scene_state(Dictionary p_state, Node *p_for_scene) {
+
 	int current = -1;
+	int index = p_state["editor_index"];
+	int n2d = 0, n3d = 0;
 	for (int i = 0; i < editor_table.size(); i++) {
```

Comment:
It would be more readable if that `for` loop (and the related `int current` declaration came just before the `if (p_state.has("editor_index"))`. The early return at line 2759 should come first as it used to.